synapse/handlers/oidc_handler.py

```diff
@@ -285,9 +285,15 @@ def __init__(
         # TODO: This is pretty much a hack to get the path specified by public_baseurl.
         # It'd probably be nicer to have a config option that lets you specify a custom
         # path, which we'd then use here.
-        self._callback_path_prefix = urlparse(hs.config.public_baseurl).path
-        if self._callback_path_prefix.endswith("/"):
-            self._callback_path_prefix = self._callback_path_prefix[:-1]
+        public_baseurl_path = urlparse(hs.config.public_baseurl).path
+        if public_baseurl_path.endswith("/"):
```

Comment:
isn't it guaranteed to end with `/`? In which case, skip this, and remove the `/` on the start of the suffix?